libres/lib/analysis/ies/ies_enkf.cpp

```diff
@@ -539,90 +434,62 @@ void ies_linalg_solve_S(const matrix_type *W0, const matrix_type *Y,
 *          S'*(S*S'+R)^{-1} H           (a)
 */
 void ies_enkf_linalg_subspace_inversion(
-    matrix_type *W0, const int ies_inversion, matrix_type *E, matrix_type *R,
-    const matrix_type *S, const matrix_type *H, double truncation,
-    double ies_steplength, int subspace_dimension, FILE *log_fp, bool dbg) {
+    matrix_type *W0, const int ies_inversion, const matrix_type *E,
+    const matrix_type *R, const matrix_type *S, const matrix_type *H,
+    double truncation, double ies_steplength, int subspace_dimension) {
 
     int ens_size = matrix_get_columns(S);
-    int m_ens_size = std::min(ens_size - 1, 16);
     int nrobs = matrix_get_rows(S);
-    int m_nrobs = std::min(nrobs - 1, 7);
-    int nrmin = std::min(ens_size, nrobs);
     double nsc = 1.0 / sqrt(ens_size - 1.0);
-    matrix_type *X1 = matrix_alloc(nrobs, nrmin); // Used in subspace inversion
-    matrix_type *X3 =
-        matrix_alloc(nrobs, ens_size); // Used in subspace inversion
-    double *eig = (double *)util_calloc(ens_size, sizeof *eig);
-
-    fprintf(log_fp, "Subspace inversion. (ies_inversion=%d)\n", ies_inversion);
+    matrix_type *X1 = matrix_alloc(
+        nrobs, std::min(ens_size, nrobs)); // Used in subspace inversion
+    std::vector<double> eig(ens_size);
 
     if (ies_inversion == IES_INVERSION_SUBSPACE_RE) {
-        fprintf(log_fp,
-                "Subspace inversion using E to represent errors. "
-                "(ies_inversion=%d)\n",
-                ies_inversion);
-        matrix_scale(E, nsc);
-        enkf_linalg_lowrankE(S, E, X1, eig, truncation, subspace_dimension);
+        matrix_type *scaledE = matrix_alloc_copy(E);
+        matrix_scale(scaledE, nsc);
+
+        enkf_linalg_lowrankE(S, scaledE, X1, eig.data(), truncation,
+                             subspace_dimension);
+
+        matrix_free(scaledE);
     } else if (ies_inversion == IES_INVERSION_SUBSPACE_EE_R) {
-        fprintf(log_fp,
-                "Subspace inversion using ensemble generated full R=EE. "
-                "(ies_inversion=%d)'\n",
-                ies_inversion);
-        matrix_scale(E, nsc);
         matrix_type *Et = matrix_alloc_transpose(E);
         matrix_type *Cee = matrix_alloc_matmul(E, Et);
         matrix_scale(
             Cee,
-            nsc *
-                nsc); // since enkf_linalg_lowrankCinv solves (SS' + (N-1) R)^{-1}
-        if (dbg)
-            matrix_pretty_fprint_submat(Cee, "Cee", "%11.5f", log_fp, 0,
-                                        m_nrobs, 0, m_nrobs);
-        enkf_linalg_lowrankCinv(S, Cee, X1, eig, truncation,
+            1.0 /
+                ((ens_size - 1) *
+                 (ens_size -
+                  1))); // since enkf_linalg_lowrankCinv solves (SS' + (N-1) R)^{-1}
```

Comment:
The formatting of this line will probably be better if the comment on the end of the line is moved to 
on top of the line. Also, that comment is quite terse so it could benefit from being expanded upon if possible.